drivers/amplifiers/ada4250/ada4250.c

```diff
@@ -151,9 +151,33 @@ int32_t ada4250_soft_reset(struct ada4250_dev *dev)
 			return ret;
 
 		if(!(data & ADA4250_RESET(ADA4250_RESET_ENABLE)))
-			return SUCCESS;
+			break;
 	}
 
+	ret = ada4250_read(dev, ADA4250_REG_GAIN_MUX, &data);
+	if (ret != SUCCESS)
+		goto error_read;
```

Comment:
nit: no need to do conditional jumps in this situation. Just return directly.

Reply:
Okay will update this